src/go/configgenerator/route_generator.go

```diff
@@ -279,32 +278,26 @@ func makeRouteTable(serviceInfo *configinfo.ServiceInfo) ([]*routepb.Route, erro
 	return backendRoutes, nil
 }
 
-func makeHttpRouteMatcher(httpRule *commonpb.Pattern) (*routepb.RouteMatch, error) {
+func makeHttpRouteMatcher(httpRule *httppattern.Pattern) (*routepb.RouteMatch, error) {
 	if httpRule == nil {
 		return nil, fmt.Errorf("httpRule is nil")
 	}
 	var routeMatcher routepb.RouteMatch
 
-	regex := httppattern.WildcardMatcherForPath(httpRule.UriTemplate)
-	if regex == "" {
-		// Match with HttpHeader method. Some methods may have same path.
+	if httpRule.UriTemplate.IsExactMatch() {
```

Comment:
Nit: If `httpRule.UriTemplate.Regex()` returns an exact match anyways, maybe no need to check and handle this case specifically.